src/elements/Label/Label.js

```diff
@@ -7,59 +7,61 @@ import {
   getUnhandledProps,
   META,
   SUI,
-  childrenUtils,
   useKeyOnly,
   useKeyOrValueAndKey,
   useValueAndKey,
 } from '../../lib'
 import { createIcon, createImage } from '../../factories'
-import { Icon, Image } from '../'
+import { Icon } from '../'
+import LabelDetail from './LabelDetail'
 
 /**
  * A label displays content classification
  */
 function Label(props) {
   const {
-    attached, basic, children, color, corner, className, circular, detail, detailAs, empty, floating, horizontal,
-    icon, image, onClick, onDetailClick, onRemove, pointing, removable, ribbon, size, tag, content,
+    attached, basic, children, circular, className, color, content, corner, detail, empty, floating, horizontal, icon,
+    image, onClick, onRemove, pointing, removable, ribbon, size, tag,
   } = props
 
   const handleClick = e => onClick && onClick(e, props)
   const handleRemove = e => onRemove && onRemove(e, props)
-  const handleDetailClick = e => onDetailClick && onDetailClick(e, props)
+
+  const pointingClass = pointing === true && 'pointing'
+      || (pointing === 'left' || pointing === 'right') && `${pointing} pointing`
+      || (pointing === 'above' || pointing === 'below') && `pointing ${pointing}`
 
   const classes = cx('ui',
     size,
     color,
+    pointingClass,
     useKeyOnly(basic, 'basic'),
     useKeyOnly(circular, 'circular'),
     useKeyOnly(floating, 'floating'),
     useKeyOnly(horizontal, 'horizontal'),
+    useKeyOnly(image === true, 'image'),
     useKeyOnly(empty, 'empty'),
     useKeyOnly(tag, 'tag'),
     useValueAndKey(attached, 'attached'),
     useKeyOrValueAndKey(corner, 'corner'),
-    useKeyOrValueAndKey(pointing, 'pointing'),
     useKeyOrValueAndKey(ribbon, 'ribbon'),
-    // TODO how to handle image child with no image class? there are two image style labels.
-    (image || childrenUtils.someByType(children, Image) || childrenUtils.someByType(children, 'img')) && 'image',
+    className,
     'label',
-    className
   )
 
-  const DetailComponent = detailAs || 'div'
   const ElementType = getElementType(Label, props)
   const rest = getUnhandledProps(Label, props)
 
+  if (children) {
+    return <ElementType {...rest} className={classes} onClick={handleClick}>{children}</ElementType>
+  }
+
   return (
     <ElementType className={classes} onClick={handleClick} {...rest}>
       {createIcon(icon)}
-      {createImage(image)}
+      {typeof image !== 'boolean' && createImage(image)}
       {content}
-      {children}
-      {detail && (
-        <DetailComponent className='detail' onClick={handleDetailClick}>{detail}</DetailComponent>
-      )}
+      {detail && <LabelDetail content={detail} />}
```

Comment:
OK, I've finished the shorthand factory update, see #529.  I also updated the Dropdown.Header shorthand I referenced above to use the new factory.

After rebasing, you can successfully use the new `createShorthand` factory.  It is meant for exactly this case, creating any arbitrary component from a shorthand value:

```jsx
{createShorthand(LabelDetail, val => ({ content: val }), detail)}
```

You specify the component, a mapValueToProps function, and then the shorthand value.